src/main/java/org/junit/contrib/java/lang/system/ExpectedSystemExit.java

```diff
@@ -138,10 +142,14 @@ public void evaluate() throws Throwable {
 
 	private void checkSystemExit() {
 		NoExitSecurityManager securityManager = (NoExitSecurityManager) getSecurityManager();
-		if (securityManager.isCheckExitCalled())
-			handleSystemExitWithStatus(securityManager.getStatusOfFirstCheckExitCall());
-		else
-			handleMissingSystemExit();
+		try {
+			if (securityManager.isCheckExitCalled())
+                handleSystemExitWithStatus(securityManager.getStatusOfFirstCheckExitCall());
+            else
+                handleMissingSystemExit();
+		} catch (InterruptedException e) {
```

Comment:
Don't catch that exception. By rethrowing it, the developer gets the full stacktrace of that exception which can be valuable. Additionally it is less code on our side.